scripts/get-usage/deduplicate-components.ts

```diff
@@ -0,0 +1,42 @@
+const componentNameChanges: Record<string, string | undefined> = {
+	'@guardian/src-ed-button/Button':
+		'@guardian/source-react-components-development-kitchen/EditorialButton',
+	'@guardian/src-ed-button/LinkButton':
+		'@guardian/source-react-components-development-kitchen/EditorialLinkButton',
+};
+
+export const deduplicateComponents = (components: string[]): string[] => {
+	const oldComponents: string[] = [];
+	const newComponents: string[] = [];
+
+	components.forEach((component) => {
+		if (component.startsWith('@guardian/src')) {
+			oldComponents.push(component);
+		} else {
+			newComponents.push(component);
+		}
+	});
+
+	const dedupedComponents: string[] = [];
+
+	oldComponents.forEach((oldComponent) => {
+		const split = oldComponent.split('/');
+		const component = split[split.length - 1];
+
+		const newPackage = oldComponent.startsWith('@guardian/src-ed')
+			? '@guardian/source-react-components-development-kitchen'
+			: '@guardian/source-react-components';
+
+		const newComponent =
+			componentNameChanges[oldComponent] ?? `${newPackage}/${component}`;
+
+		if (newComponents.includes(newComponent)) {
+			dedupedComponents.push(`${oldComponent} | ${newComponent}`);
+			newComponents.splice(newComponents.indexOf(newComponent), 1);
+		} else {
+			dedupedComponents.push(oldComponent);
+		}
+	});
+
+	return [...dedupedComponents, ...newComponents];
+};
```

Comment:
I'm a little worried that the complexity of this function could become unsustainable as components are moved around between the Kitchen and stable Source. For instance if we choose to demote Footer to the Kitchen, would we add lines to this function to try and group usage of this component in stable Source and in the Kitchen?

If this functionality is useful now, we should probably put a shelf life on it.

Reply:
To summarise the discussion we had offline:

That's a really good point and something that I was slightly concerned about, even without having fully comprehended the complexity it would introduce. If we did make that change and felt we needed to introduce some further grouping, I do think it would be possible, but maybe a preferable scenario is that we encourage speedy migration and then drop monitoring for `src-*` components so that we don't have to shoulder both bits of complexity at the same time.